File: tasks/vulcancyber-v1/oss-werkzeug-etag-strict-parse/tests/vb_regression.py
# Hidden pass_to_pass regression guard for oss-werkzeug-etag-strict-parse.
#
# Rejecting invalid ETags must not change parsing of well-formed ones: valid quoted
# ETags are still parsed and still match, and a valid ETag matches by value. Both
# hold at the base commit and after the fix. Run with pytest.

from werkzeug.http import parse_etags


def test_valid_quoted_etags_parsed():
    etags = parse_etags('"a", "b"')
    assert etags.as_set() == {"a", "b"}


def test_valid_etag_matches():
    etags = parse_etags('"a", "b"')
    assert "a" in etags
    assert "missing" not in etags

File: tasks/vulcancyber-v1/oss-quick-xml-serialize-control-escape/tests/tests/vb_regression.rs
// Hidden pass_to_pass regression guard for oss-quick-xml-serialize-control-escape.
//
// Escaping additional control characters must not change how the ordinary XML
// metacharacters are escaped, nor how a plain attribute is written. Both hold at
// the base commit and after the fix. Run with `cargo test --offline`.

use quick_xml::escape::escape;
use quick_xml::events::{BytesStart, Event};
use quick_xml::Writer;

#[test]
fn ordinary_metacharacters_still_escaped() {
    let escaped = escape("a<b>&c\"d'e");
    assert!(escaped.contains("&lt;"));
    assert!(escaped.contains("&gt;"));
    assert!(escaped.contains("&amp;"));
    // The five XML predefined entities must be unchanged; no literal <, >, & remain.
    assert!(!escaped.contains('<'));
    assert!(!escaped.contains('>'));
    assert!(!escaped.contains('&') || escaped.contains("&amp;"));
}

#[test]
fn plain_attribute_value_is_written_verbatim() {
    let mut writer = Writer::new(Vec::new());
    let mut elem = BytesStart::new("e");
    elem.push_attribute(("k", "plain-value"));
    writer.write_event(Event::Start(elem)).unwrap();
    let out = String::from_utf8(writer.into_inner()).unwrap();
    assert!(out.contains("k=\"plain-value\""), "got {out:?}");
}
